src/metadata_validator.py

```diff
@@ -322,6 +345,23 @@ def validate_required_props(self, data_record, msg_prefix):
             if anode_definition["properties"][data_key]["required"]:
                 if data_value is None or not str(data_value).strip():
                     result[ERRORS].append(create_error("M003",[msg_prefix, data_key], data_key, data_value))
+                else:
+                    # validate program name and study name required in CRDCDH-2431.  Both are required properties.
+                    if node_type == "program" and data_key == "program_name":
```

Comment:
We should use entityType instead of node type, because node type can be different in different data commons. Same for the property name, we probably need to add some new configurations in model repo's content.json to configure property names.